include/aws/common/trace_event.h

```diff
@@ -0,0 +1,84 @@
+#ifndef AWS_COMMON_TRACE_EVENT_H
+#define AWS_COMMON_TRACE_EVENT_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+/* Disable warnings with windows build */
+#ifdef _MSC_VER
+#    pragma warning(disable : C4996)
+#endif
```

Comment:
Don't disable warnings from header files.
That causes the warnings to be disabled in ANY code that includes this header, including our customer's code.
Our customer might not want such a warning disabled.

move this to the .c file

A big difference between student/hobby projects and giant corporate codebases is:
In small code bases you put stuff in headers to simplify writing your .c files.
In big corporate codebases you only put what's absolutely necessary in your headers. You don't want to add anything to your user's .c files that they might not want (extra includes they don't necessarily need that increase compile time, disabled warnings they might not want disabled, stuff like that)